src/arc-mutex/arc-drop.md

```diff
@@ -89,9 +89,13 @@ Now, let's wrap this all up inside the `Drop` implementation:
 ```rust,ignore
 impl<T> Drop for Arc<T> {
     fn drop(&mut self) {
-        let inner = unsafe { self.ptr.as_ref() };
-        if inner.rc.fetch_sub(1, Ordering::Release) != 1 {
-            return;
+        // We create a scope so that inner won't be live
+        // when we create a Box to drop the `ArcInner`
+        {
+            let inner = unsafe { self.ptr.as_ref() };
+            if inner.rc.fetch_sub(1, Ordering::Release) != 1 {
+                return;
+            }
```

Comment:
Hm, if this is indeed not safe I believe that my PR does not completely solve the problem. Another thread can intersperse itself after `rc` has been decreased do the whole drop glue before we leave the scope. In this case, I think that any reference should be avoided.